plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1169,26 +1177,37 @@ private String getDataStoreUrlFromStore(DataStoreTO store) {
         return store.getUrl();
     }
 
-    protected synchronized String attachOrDetachDevice(final Connect conn, final boolean attach, final String vmName, final String xml) throws LibvirtException, InternalErrorException {
+    protected synchronized String attachOrDetachDevice(final Connect conn, final boolean attach, final String vmName, final String xml)
+            throws LibvirtException, InternalErrorException {
         Domain dm = null;
         try {
             dm = conn.domainLookupByName(vmName);
 
             if (attach) {
                 s_logger.debug("Attaching device: " + xml);
                 dm.attachDevice(xml);
-            } else {
-                s_logger.debug("Detaching device: " + xml);
-                dm.detachDevice(xml);
-                LibvirtDomainXMLParser parser = new LibvirtDomainXMLParser();
-                parser.parseDomainXML(dm.getXMLDesc(0));
-                List<DiskDef> disks = parser.getDisks();
-                for (DiskDef diskDef : disks) {
-                    if (StringUtils.contains(xml, diskDef.getDiskPath())) {
-                        throw new InternalErrorException("Could not detach volume. Probably the VM is in boot state at the moment");
-                    }
+                return null;
```

Comment:
better throw an exception or a specific error value. this is a bad pattern.

Reply:
In the attach workflow it was already like this. I am not following what you are saying that it is a bad practice. What would you expect to return here? We are focusing on the detach method, to handle the situations when the volume takes a bit longer to be removed from the VM (e.g. the disk has some writes being written, and KVM waits until the write is finished).

Reply:
@RodrigoDLopez returning null is a bad practice. it can lead to NPEs in unexpected places. Better to throw a specific exception and hendle it where appropriate.
You are right that this pattern has been implemented a lot of times but we shouldn´t add to this.

Reply:
@RodrigoDLopez after closer inspection; this method does not return anything useful anywhere. There are two `return` statements both returning `null`. Letś make this a `void` method.